sklbench/datasets/loaders.py

```diff
@@ -0,0 +1,809 @@
+
+
+@cache
+def load_susy(**kwargs) -> Tuple[Dict, Dict]:
+    return load_higgs_susy_subsample(**kwargs)
+
+
+@cache
+def load_letters(
+    data_name: str, data_cache: str, raw_data_cache: str, dataset_params: Dict
+) -> Tuple[Dict, Dict]:
+    """
+    Letter Recognition dataset from UCI machine learning repository
+    http://archive.ics.uci.edu/ml/datasets/Letter+Recognition
+
+    Classification task. n_classes = 26.
+    """
+    url = (
+        "http://archive.ics.uci.edu/ml/machine-learning-databases/"
+        "letter-recognition/letter-recognition.data"
+    )
+    data = download_and_read_csv(url, raw_data_cache, header=None, dtype=None)
+    x, y = data.iloc[:, 1:], data.iloc[:, 0].astype("category").cat.codes.values
+
+    data_desc = {"n_classes": 26, "default_split": {"test_size": 0.2, "random_state": 0}}
+    return {"x": x, "y": y}, data_desc
+
+
+@cache
+def load_sklearn_digits(
+    data_name: str, data_cache: str, raw_data_cache: str, dataset_params: Dict
+) -> Tuple[Dict, Dict]:
+    x, y = load_digits(return_X_y=True)
+    data_desc = {
+        "n_classes": 10,
+        "default_split": {"train_size": 0.2, "random_state": 42},
+    }
+    return {"x": x, "y": y}, data_desc
+
+
+@cache
+def load_covtype(
+    data_name: str, data_cache: str, raw_data_cache: str, dataset_params: Dict
+) -> Tuple[Dict, Dict]:
+    """
+    Cover type dataset from UCI machine learning repository
+    https://archive.ics.uci.edu/ml/datasets/covertype
+
+    y contains 7 unique class labels from 1 to 7 inclusive.
+    Classification task. n_classes = 7.
+    """
+    x, y = fetch_covtype(return_X_y=True, data_home=raw_data_cache)
+    y = y.astype(int) - 1
+    binary = dataset_params.get("binary", False)
+    if binary:
+        y = (y > 2).astype(int)
+
+    data_desc = {
+        "n_classes": 2 if binary else 7,
+        "default_split": {"test_size": 0.2, "random_state": 77},
+    }
+    return {"x": x, "y": y}, data_desc
+
+
+@cache
+def load_epsilon(
+    data_name: str, data_cache: str, raw_data_cache: str, dataset_params: Dict
+) -> Tuple[Dict, Dict]:
+    """
+    Epsilon dataset
+    https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary.html
+
+    Classification task. n_classes = 2.
+    """
+    url_train = (
+        "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary"
+        "/epsilon_normalized.bz2"
+    )
+    url_test = (
+        "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary"
+        "/epsilon_normalized.t.bz2"
+    )
+    local_url_train = os.path.join(raw_data_cache, os.path.basename(url_train))
+    local_url_test = os.path.join(raw_data_cache, os.path.basename(url_test))
+
+    retrieve(url_train, local_url_train)
+    retrieve(url_test, local_url_test)
+
+    x_train, y_train = load_svmlight_file(local_url_train, dtype=np.float32)
+    x_test, y_test = load_svmlight_file(local_url_test, dtype=np.float32)
+
+    x = sparse.vstack([x_train, x_test])

```

Comment:
is there a reason sparse is used instead of np.vstack here?

Reply:
x_train and x_test are sparse.